driver/task.go

```diff
@@ -249,16 +265,19 @@ func (t *Task) Source() string {
 	return t.source
 }
 
-// VariableFiles returns a copy of the list of configured variable files
-// for the task's module.
+// VariableFiles returns a copy of the configured variable file
+// for the task's module as a list.
 func (t *Task) VariableFiles() []string {
 	t.mu.RLock()
 	defer t.mu.RUnlock()
 
-	varFiles := make([]string, len(t.varFiles))
-	for i, vf := range t.varFiles {
-		varFiles[i] = vf
+	// In the future may be worth refactoring for Terraform CLI to only expect a
+	// single varFile, for now this is useful
```

Comment:
Changing existing behavior that Terraform CLI and CTS supports would be breaking change that doesn't offer additional value. Could the API support multiple files instead?

It provides users more flexibility to organize their variables, like one file contains more sensitive information, and the other could act more like parameters.

Reply:
So CTS still supports multiple file input, it just writes out a single file to the working directory, and that single file is fed to terraform CLI. So from a user perspective they can still organize the files any way they want eg.

var_file_1  ------>
var_file_2 ------>       variables.tfvars
var_file_3 ------>

Reply:
So I don't see this as a breaking change I guess, nothing changes in the configuration or input to CTS

Reply:
Based on the example request in the PR summary, it seems like this is the beginning of diverging UX between config file and API where only one file is supported, and this comment leads to suggest breaking change. Which way is it?

```json
     "variables":{
        "filename": "\"set.txt\"",
        "object": "..."
      }
```

Reply:
maybe a poor example choice on my part, I had built an example module on top of the simple example module that's part of the CTS repo. Filename in this context doesn't indicate a file for input, it's a key-value (it's the name of the file that the terraform module creates and write service information to)

The API just support key-values, it doesn't have any concept of files

Reply:
Okay that example was pretty confusing especially in addition to changing the `Task.varFiles` to a singular file. Have you double checked if this change behaves correctly for the existing use-case for loading a [task from the config file that contains multiple variable files](https://www.consul.io/docs/nia/configuration#variable_files). This now gets ignored https://github.com/hashicorp/consul-terraform-sync/blob/main/driver/task.go#L82

Reply:
Chatted and this use-case is covered with the existing code path that loads variables from the files during initialization. A follow-up non-blocking refactor would be to look into how the `variables.module.tf` file is generated and couple it to this new direction for its `.tfvars`. Maybe the `TaskConfig.varFiles`  could be removed completely at this point.